File: Cpp/Binary search/Find Kth Rotation.cpp
//https://www.geeksforgeeks.org/problems/rotation4723/1
class Solution {
  public:
    int findKRotation(vector<int> &a) {
        // Code Here
        int n=a.size();
        int l=0, h=n-1, ans = 0;
        while(l<=h){
            int mid = l+((h-l)/2);
            int i = (mid-1+n)%n;
            int j = (mid+1)%n;
            if(a[mid]<a[i] && a[mid]<a[j]){
                return mid;
            }
            if(a[h]>a[mid]){
                h=mid-1;
            }else{
                ans = mid;
                l=mid+1;
            }
        }
        return ans;
        
    }
};

File: Python/Array/Find Kth Rotation.py
//https://www.geeksforgeeks.org/problems/rotation4723/1
class Solution:
    def findKRotation(self, arr):
        # code here
        n = len(arr)
        l, h = 0, n - 1
        ans = 0
        
        while l <= h:
            mid = l + (h - l) // 2
            i = (mid - 1 + n) % n
            j = (mid + 1) % n
            
            if arr[mid] < arr[i] and arr[mid] < arr[j]:
                return mid
            
            if arr[h] > arr[mid]:
                h = mid - 1
            else:
                ans = mid
                l = mid + 1
        
        return ans
